### ISSUE TITLE
Remove the deprected calls to cudaConfigureCall / cudaSetupArgument / cudaLaunch  

### ISSUE THREAD

**Issue by AAAlvesJr** [jaredhoberock/bulk#47]
Dear Jared, there is any chance of updating the triple_chevron_launcher and remove the calls for the deprected (now removed) CUDA Execution Control API?
We use this bulk + thrust on Hydra here at CERN, and would like to keep running it with the newer versions of CUDA.  
Cheers

**Comment by jaredhoberock** [comment]
It seems like the examples no longer compile with recent CUDA Toolkits, but I've put a potential fix in this branch:

    https://github.com/jaredhoberock/bulk/tree/issue-47

I'm unable to actually test the fix on my workstation due to the bit rot. Could you try it and let me know if it works for you?

**Comment by AAAlvesJr** [comment]
Hi Jared, thanks for the prompt reaction. The template parameter Args in ``reinterpret_cast<void*>(const_cast<Args*>(&task))`` is not defined:

bulk/detail/cuda_launcher/triple_chevron_launcher.hpp(128): error: identifier "Args" is undefined
bulk/detail/cuda_launcher/triple_chevron_launcher.hpp(185): error: identifier "Args" is undefined

On the other hand the errors connected to calling the absend API functions went way.  
So, in conclusion, i could not test the patch completely.

**Comment by jaredhoberock** [comment]
Thanks for testing. I've corrected the typo, could you try again?

**Comment by AAAlvesJr** [comment]
Hi Jared,  changing from 
```
 bulk::detail::throw_on_error(
            cudaLaunchKernel(super_t::global_function_pointer(), dim3(num_blocks), dim3(block_size), pointers_to_args, num_dynamic_smem_bytes, stream),
            "after cudaLaunchKernel in triple_chevron_launcher::launch()"
          );
```
to 

```
 bulk::detail::throw_on_error(
            cudaLaunchKernel(reinterpret_cast<void*>(super_t::global_function_pointer()), dim3(num_blocks), dim3(block_size), pointers_to_args, num_dynamic_smem_bytes, stream),
            "after cudaLaunchKernel in triple_chevron_launcher::launch()"
          );
```
solve the compilation problems. Thank you very much

**Comment by AAAlvesJr** [comment]
So, turns out that while I could get the compilation problems solved,  I started facing a **segmentation violation** when processing functions/parameters larger than 4096 B with CUDA 10.1. This does not happen if I use the previous versions of CUDA. I will try to write down an example you can run on your side, my current use case is too complicated to show the problem in a concise way.  What I did was to update bulk version of thrust v1.8.9 to make it compatible with CUDA 10.1, then I integrated it on [Hydra.](https://github.com/MultithreadCorner/Hydra) . Everything works fine, as far as the user does not try to pass to complex functions, like happens in the phys/dalitz_plot example. 
Any idea what can be going on? 

Thanks

**Comment by jaredhoberock** [comment]
Hmm, hard to say without a reproducer. That said, this repository is essentially deprecated and unfortunately I don't have the bandwidth to debug issues like this. I'm happy to integrate any fixes you uncover, though.

Are you using Bulk directly within Hydra, or are you simply using Thrust? There ought to be a newer version of Thrust available that doesn't use Bulk in its implementation at all.

**Comment by AAAlvesJr** [comment]
I am trying to write down one reproducer.  Thank you for the good will.

I am not using Bulk directly, but through Thrust. In physics, complex 
and large functions are very common, then the feature of Bulk of 
providing a alternative patch for dispatching of such larger functions 
in CUDA backend is very important for Hydra.

The new version of Thrust, being based in CUB for CUDA backend, does not 
have such feature. From version 7.5 to version 9.2, all this worked like 
a charm. With the newer versions of Thrust (v1.9.x) this does not even 
compile.

For example, errors like this one show up in Thrust v1.9.x, with CUDA 10.1

```

cub/device/dispatch/dispatch_reduce.cuh(113): Error: Formal parameter 
space overflowed (4605 bytes required, max 4096 bytes allowed) in 
function ...

```

Em 02/10/2019 22:28, Jared Hoberock escreveu:
